astropy/coordinates/sky_coordinate.py

```diff
@@ -1888,17 +1900,33 @@ def guess_from_table(cls, table, **coord_kwargs):
             ends_with_comp = r'.*(\W|\b|_)' + comp_name + r'\b'
             # the final regex ORs together the two patterns
             rex = re.compile('(' + starts_with_comp + ')|(' + ends_with_comp + ')',
-                             re.IGNORECASE | re.UNICODE)
-
-            for col_name in table.colnames:
-                if rex.match(col_name):
-                    if comp_name in comp_kwargs:
-                        oldname = comp_kwargs[comp_name].name
-                        msg = ('Found at least two matches for  component "{0}"'
-                               ': "{1}" and "{2}". Cannot continue with this '
-                               'ambiguity.')
-                        raise ValueError(msg.format(comp_name, oldname, col_name))
-                    comp_kwargs[comp_name] = table[col_name]
+              re.IGNORECASE | re.UNICODE)
+
+            # find all matches
+            matches = [col_name for col_name in table.colnames
+                       if rex.match(col_name)]
+
+            # now need to select among matches, also making sure we don't have
+            # an exact match with another component
+            if len(matches) == 0:  # no matches
+                continue
+            elif len(matches) == 1:  # only one match
+                col_name = matches[0]
+            else:  # more than 1 match
+                # try to sieve out other components
+                [matches.remove(n) for n in representation_component_names
+                 if (n in matches and n != comp_name)]
+
+                # if there's only one remaining match, it worked.
+                if len(matches) == 1:
+                    col_name = matches[0]
+                else:
+                    msg = (
+                       'Found at least two matches for  component "{0}"'
+                       ': "{1}". Cannot continue with this ambiguity.')
```

Comment:
I know this was in the original version, but maybe this should be a smidge clearer here, like:
```suggestion
                       ': "{1}". Cannot guess coordinates from a table with this ambiguity.')
```
or similar?